ludwig/models/trainer.py

```diff
@@ -91,40 +94,84 @@ def __exit__(self, exc_type, exc_val, exc_tb):
         self.shutdown()
 
 
+@dataclass
+class TrainerConfig:
+    optimizer: Dict = schema.OptimizerOptions(default={TYPE: "adam"})
+    epochs: int = schema.PositiveInteger(default=100)
```

Comment:
Totally not required for this PR -- this seems like an appropriate place to document what all of the different parameters mean, at least in comments. WDYT?

Reply:
Agreed! Going to do this in the next couple of commits.